File: lab8quadratic.py
import math

class QuadraticEquationSolver:
    
    def findRoots(self, a, b, c):

        discriminant = b**2 - 4*a*c
        
        if discriminant < 0:
            return "No Real Roots"
        
        root1 = (-b + math.sqrt(discriminant)) / (2 * a)
        root2 = (-b - math.sqrt(discriminant)) / (2 * a)
        
        return root1, root2

solver = QuadraticEquationSolver()
a = float(input("Enter coefficient a: "))
b = float(input("Enter coefficient b: "))
c = float(input("Enter coefficient c: "))

roots = solver.findRoots(a, b, c)
print("The roots of the equation are:", roots)

File: lab8quadratic.java
public class lab8quadratic {
    
    public static String findRoots(double a, double b, double c) {
        // Calculate discriminant
        double discriminant = b * b - 4 * a * c;
        
        // If discriminant is negative, roots are imaginary
        if (discriminant < 0) {
            return "No Real Roots";
        }
        
        // Calculate the two roots
        double root1 = (-b + Math.sqrt(discriminant)) / (2 * a);
        double root2 = (-b - Math.sqrt(discriminant)) / (2 * a);
        
        return "The roots are: " + root1 + " and " + root2;
    }

    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        
        System.out.print("Enter coefficient a: ");
        double a = scanner.nextDouble();
        
        System.out.print("Enter coefficient b: ");
        double b = scanner.nextDouble();
        
        System.out.print("Enter coefficient c: ");
        double c = scanner.nextDouble();
        
        System.out.println(findRoots(a, b, c));
        scanner.close();
    }
}

